.github/workflows/ci.yml

```diff
@@ -0,0 +1,45 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version : '3.13'
+
+      - name: Install dependencies
+        run: |
+          python -m venv .venv
+          source .venv/bin/activate
+          pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Format code
+        run: |
+          source .venv/bin/activate
+          mypy src 
+
+      - name: Lint code
+        run: |
+          source .venv/bin/activate
+          flake8 src tests
+
+      - name: Run tests
+        run: |
+          source .venv/bin/activate
+          pytest
```

Comment:
specify the target folder of the tests and add a new line at the end of the file